docs/content/topics/scripting/guide.md

```diff
@@ -9,1401 +9,867 @@ aliases:
   - /topics/scripting/guide.md
 ---
 
-TODO: update per v2
-
 # Scripting Guide
 
-This guide explains the basics of how to create and structure `brigade.js` files.
+This guide explains the basics of how to create and structure Brigade scripts,
+which can be written in either JavaScript (`brigade.js`) or TypeScript
+(`brigade.ts`).
 
 # Brigade Scripts, Projects, and Repositories
 
-Brigade scripts are stored in a `brigade.js` file. They are designed to contain short scripts
-that coordinate running multiple related tasks. We like to think of these as
-cluster-oriented shell scripts: The script just ties together a bunch of other programs.
+Brigade scripts can be stored either in a project definition or in a project's
+git repository. They are designed to organize and coordinate running multiple
+related tasks, usually in response to events. We like to think of these as
+cluster-oriented shell scripts: The script just ties together a bunch of other
+programs.
 
-At the very core, a Brigade script is simply a JavaScript file that is executed within a
-special cluster environment. An environment is composed of the following things:
+At the very core, a Brigade script is simply a JavaScript (or TypeScript) file
+that is executed within a special cluster environment. An environment is
+composed of the following things:
 
-- A Brigade server running in your cluster
-- A [project](./projects.md) in which the script will run
+- A Brigade server [running in your Kubernetes cluster][quickstart]
+- A [project] in which the script will run
 - (Optionally) A source code repository (e.g. git) that contains data our
-- (Optionally) A [`brigade.json`](workers.md) file that contains additional dependencies that can be used from `brigade.js`
-- (Optionally) Other configuration data, like [secrets](secrets.md)
+  script may utilize
+- (Optionally) A [`package.json`] file that contains additional dependencies
+  that can be used in the script
+- (Optionally) Other configuration data, like [secrets]
```

Comment:
I'd consider simplifying this section. Some aspects seem inaccurate in nuanced ways. (For instance, "A project in which the script will run" almost implies a project is a runtime environment like a container.) Rather than spend energy rewording, I'm wondering if understanding of what a Brigade script is can be conveyed without this level of detail.

Would it be enough to state that the Brigade script is the JavaScript or TypeScript is defined by a Brigade project and that Brigade executes that script in the context of a worker to handle events to which the project is subscribed?